app/controllers/merchant/base_controller.rb

```diff
@@ -0,0 +1,7 @@
+class Merchant::BaseController < ApplicationController
+  before_action :require_merchant
+
+  def require_merchant
+    render file: "/public/404" unless current_merchant?
+  end
+end
```

Comment:
This answers my other question about if we had a merchant base controller, I'll resolve that comment now.